docs/operational_guide/bootstrapping.md

```diff
@@ -0,0 +1,106 @@
+
+| Node | Shard | State |
+|------|-------|-------|
+|  1   |   1   |   I   |
+|  1   |   2   |   I   |
+|  1   |   3   |   I   |
+|  2   |   1   |   I   |
+|  2   |   2   |   I   |
+|  2   |   3   |   I   |
+|  3   |   1   |   A   |
+|  3   |   2   |   A   |
+|  3   |   3   |   A   |
+
+In this case, the peer bootstrapper running on node 1 will not be able to fullfill any requests because node 2 is in the Initializing state for all of its shards and cannot fulfill bootstrap requests. This means that node 1's peer bootstrapper cannot meet its default consistency level of majority for bootstrapping (1 < 2 which is majority with a replication factor of 3). On the other hand, node 1 would be able to peer bootstrap in the following topology because its peers (nodes 2/3) are available for all of their shards:
+
+| Node | Shard | State |
+|------|-------|-------|
+|  1   |   1   |   I   |
+|  1   |   2   |   I   |
+|  1   |   3   |   I   |
+|  2   |   1   |   A   |
+|  2   |   2   |   A   |
+|  2   |   3   |   A   |
+|  3   |   1   |   A   |
+|  3   |   2   |   A   |
+|  3   |   3   |   A   |
+
+Note that a bootstrap consistency level of majority is the default value, but can be modified by changing the value of the key "m3db.client.bootstrap-consistency-level" in EtcD to one of: "none", "one", "unstrict_majority" (attempt to read from majority, but settle for less if any errors occur), "majority" (strict majority), and "all".
+
+**Note**: Any bootstrappers configuration that does not include the peers bootstrapper will be unable to handle dynamic topology changes of any kind.
+
+### Uninitialized Topology Bootstrapper
+
+The purpose of the uninitialzied topology bootstrapper is to succeed bootstraps for all time ranges for shards that have never been completely bootstrapped (at a cluster level). This allows us to run the default bootstrapper configuration of: `filesystem,commitlog,peers,topology_uninitialized` such that the filesystem and commitlog bootstrappers are used by default in node restarts, the peer bootstrapper is used for node adds/removes/replaces, and bootstraps still succeed for brand new topologies where both the commitlog and peers bootstrappers will be unable to succeed any bootstraps. In other words, the uninitialized topology bootstrapper allows us to place the commitlog bootstrapper *before* the peers bootstrapper and still succeed bootstraps with brand new topologies without resorting to using the noop-all bootstrapper which suceeds bootstraps for all shard/time-ranges regardless of the status of the topology.
+
+The uninitialized topology bootstrapper determines whether a topology is "new" for a given shard by counting the number of hosts in the Initializing state and Leaving states and if the number of Initializing - Leaving > 0 than it succeeds the bootstrap because that means the topology has never reached a state where all hosts are Available.
```

Comment:
`if the number of Initializing - Leaving > 0 than it succeeds` -> `if there are more Initializing than Leaving, then it succeeds`